components/multi-attribute-login/org.wso2.carbon.identity.unique.claim.mgt/src/main/java/org/wso2/carbon/identity/unique/claim/mgt/listener/UniqueClaimUserOperationEventListener.java

```diff
@@ -435,4 +441,30 @@ public boolean doPreUpdateCredentialByAdmin(String userName, Object newCredentia
         validatePasswordNotEqualToClaims(claimMap, userStoreManager, newCredential);
         return true;
     }
+
+    private boolean isMultiValuedClaimDuplicated(String username, String claimUri, String claimValue, String profile,
+                                                 UserStoreManager userStoreManager, String domainName,
+                                                 ClaimConstants.ClaimUniquenessScope uniquenessScope)
+            throws UserStoreException {
+
+        String multiAttributeSeparator = userStoreManager.getSecondaryUserStoreManager(domainName)
+                .getRealmConfiguration().getUserStoreProperty(MULTI_ATTRIBUTE_SEPARATOR);
+
+        String[] userList;
+        for (String claimValuePart : claimValue.split(multiAttributeSeparator)) {
+            if (ClaimConstants.ClaimUniquenessScope.WITHIN_USERSTORE.equals(uniquenessScope)) {
+                String claimValueWithDomain = domainName + UserCoreConstants.DOMAIN_SEPARATOR + claimValuePart;
+                userList = userStoreManager.getUserList(claimUri, claimValueWithDomain, profile);
+            } else {
+                userList = userStoreManager.getUserList(claimUri, claimValue, profile);
+            }
+            if (userList.length == 1) {
+                String usernameWithUserStoreDomain = UserCoreUtil.addDomainToName(username, domainName);
+                return !usernameWithUserStoreDomain.equalsIgnoreCase(userList[0]);
+            } else {
+                return userList.length != 0;
+            }
+        }
+        return false;
+    }
```

Comment:
Need to check whether uniqueness validation improvement is required or not for the multi valued claims.

For a cases of update for a multi valued claim where the uniqueness enabled for that claim, if it contains large number of elements, the uniqueness check will take longer time as it iterate over each attribute and check the uniqueness.